core/lib/spree/testing_support/common_rake.rb

```diff
@@ -36,6 +36,7 @@
     ]
 
     puts 'Setting up dummy database...'
+    system('bin/rails db:environment:set RAILS_ENV=test')
```

Comment:
The `test_app` raises an error asking to set the db environment right after some migration fails.